spec/components/create_a_job_sidebar_component_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe CreateAJobSidebarComponent, type: :component do
+  let(:vacancy) { create(:vacancy, completed_step: completed_step) }
+  let(:completed_step) { 0 }
+  let(:current_step) { 1 }
+  let(:school_group_user?) { true }
+
+  before do
+    allow_any_instance_of(HiringStaff::JobCreationHelper).to receive(:current_step).and_return(current_step)
+    allow_any_instance_of(HiringStaff::JobCreationHelper).to receive(:school_group_user?).and_return(school_group_user?)
+  end
+
+  it 'renders the sidebar' do
+    render_inline(described_class.new(vacancy: vacancy))
+
+    expect(rendered_component).to include('Create a job listing steps')
+  end
+
+  it 'renders active class on current_step' do
+    result = render_inline(described_class.new(vacancy: vacancy))
+
+    assert_includes(
+      result.css('.app-step-nav__step--active .app-step-nav__circle-background').to_html, current_step.to_s
+    )
+  end
+
+  context 'when a step is completed' do
+    let(:completed_step) { 1 }
+    let(:current_step) { 2 }
+
+    it 'renders visited class on completed steps' do
+      result = render_inline(described_class.new(vacancy: vacancy))
+
+      assert_includes(
+        result.css('.app-step-nav__step--visited .app-step-nav__circle-background').to_html, completed_step.to_s
+      )
+    end
+  end
+
+  context 'when a School user creates a job' do
+    let(:school_group_user?) { false }
+
+    it 'renders the correct steps' do
+      render_inline(described_class.new(vacancy: vacancy))
+
+      expect(rendered_component).not_to include(I18n.t('jobs.job_location'))
+      expect(rendered_component).to include(I18n.t('jobs.job_details'))
+      expect(rendered_component).to include(I18n.t('jobs.pay_package'))
+      expect(rendered_component).to include(I18n.t('jobs.important_dates'))
+      expect(rendered_component).to include(I18n.t('jobs.supporting_documents'))
+      expect(rendered_component).to include(I18n.t('jobs.application_details'))
+      expect(rendered_component).to include(I18n.t('jobs.job_summary'))
+      expect(rendered_component).to include(I18n.t('jobs.review_heading'))
```

Comment:
There might be a case for using a regex match assertion, if you want to test the order of these inclusions. There is already some order-checking in `job_creation_helper_spec.rb` though.